src/scenes/Engagement/EditEngagement/EditEngagementDialog.tsx

```diff
@@ -253,6 +253,45 @@ export const EditEngagementDialog: FC<EditEngagementDialogProps> = ({
       }}
       {...props}
       initialValues={initialValues}
+      validate={(values) => {
+        const start = values.engagement.startDateOverride;
+        const end = values.engagement.endDateOverride;
+
+        if (start && end && start > end) {
+          return {
+            engagement: {
+              startDateOverride: 'Start date should come before end date',
+              endDateOverride: 'End date should come after start date',
+            },
+          };
+        }
+
+        if (
+          start &&
```

Comment:
`end` could still be set here. Either check it's null again or nest the if statement above and return if start < end.